pkg/sql/opt/cat/column.go

```diff
@@ -43,6 +58,9 @@ type Column interface {
 	// than every column allocated before or after. This is true even if a column
 	// is dropped and then re-added with the same name; the new column will have
 	// a different ID. See the comment for StableID for more detail.
+	//
+	// Virtual columns don't have stable IDs; for these columns ColID() should not
```

Comment:
I'm also ok saying `must not be called` instead of should here and below, to strengthen to guarantees.